### PR TITLE
WFLY-5922 Don't re-export modules from the EJB API module

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
Retest this please

File: feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml [hunk 1]
```diff
@@ -25,11 +25,11 @@
 <module xmlns="urn:jboss:module:1.3" name="javax.ejb.api">
     <dependencies>
         <!-- This dep is for javax.naming -->
-        <module name="javax.api" export="true"/>
-        <module name="javax.transaction.api" export="true"/>
```
Comment (wolfc) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: Shouldn't this one be kept as it is a hard dependency?

File: feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml [hunk 2]
```diff
@@ -25,11 +25,11 @@
...
-        <module name="javax.xml.rpc.api" export="true"/>
```
Comment (wolfc) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: Likewise?

Comment (stuartwdouglas) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: In general 'export' just causes problems, no other modules export API's that they depend on. These other modules will still be added to deployments as a dependency, but they can now be excluded if needed.

Comment (wolfc) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: Isn't that really a problem with the exclusion functionality? If I want to exclude a module I would expect it to exclude any transitive ones as well.

Comment (bstansberry) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: So if the user declares an exclusion for "javax.ejb.api" we automagically add exclusions for any module it exports?

Possible, but then if they actually want, say "javax.transaction.api", they need to configure a "don't-exclude". After we develop such a config of course. ;)

Comment (scottmarlow) [inline feature-pack/src/main/resources/modules/system/layers/base/javax/ejb/api/main/module.xml]: where should we be exporting the "javax.xml.rpc.api" module to deployments instead?  We are failing the signature test for jax-rpc.

**Comment by n1hility** [comment]
@stuartwdouglas @bstansberry @wolfc as a followup java.xml.rpc.api and javax.transaction.api are hard  linked deps and should have been kept as exported deps. @bstansberry has a fix that adds imports on the other subsystem bits, which is good enough for now, but we should readd those to post 10 final

**Pull request merged**